awswrangler/opensearch/_write.py

```diff
@@ -0,0 +1,471 @@
+            ):
+                try:
+                    v = json.loads(v)
+                except Exception as e:
+                    try:
+                        v = ast.literal_eval(v)  # if properties are enclosed with single quotes
+                    except:
+                        _logger.warning(f'could not convert string to json: {v}')
+                        _logger.warning(e)
+        return v
+
+    df_iter = df.iterrows()
+    for i, document in df_iter:
+        # print(document)
+        # yield document
+        yield {k: _deserialize(v) for k, v in document.items() if notna(v)}
+
+
+def _file_line_generator(path: str, is_json: bool = False):
+    with open(path) as fp:
+        for line in fp:
+            if is_json:
+                yield json.loads(line)
+            else:
+                yield line.strip()
+
+
+def create_index(
+    client: Elasticsearch,
+    index: str,
+    doc_type: Optional[str] = None,
+    settings: Optional[Dict[str, Any]] = None,
+    mappings: Optional[Dict[str, Any]] = None
+) -> Dict[str, Any]:
+    """Creates an index.
+
+    Parameters
+    ----------
+    client : Elasticsearch
+        instance of elasticsearch.Elasticsearch to use.
+    index : str
+        Name of the index.
+    doc_type : str, optional
+        Name of the document type (for Elasticsearch versions 5.x and earlier).
+    settings : Dict[str, Any], optional
+        Index settings
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#index-settings
+    mappings : Dict[str, Any], optional
+        Index mappings
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#mappings
+
+    Returns
+    -------
+    Dict[str, Any]
+        OpenSearch rest api response
+        https://opensearch.org/docs/opensearch/rest-api/create-index/#response.
+
+    Examples
+    --------
+    Creating an index.
+
+    >>> import awswrangler as wr
+    >>> client = wr.opensearch.connect(host='DOMAIN-ENDPOINT')
+    >>> response = wr.opensearch.create_index(
+    ...     client=client,
+    ...     index="sample-index1",
+    ...     mappings={
+    ...        "properties": {
+    ...          "age":  { "type" : "integer" }
+    ...        }
+    ...     },
+    ...     settings={
+    ...         "index": {
+    ...             "number_of_shards": 2,
+    ...             "number_of_replicas": 1
+    ...          }
+    ...     }
+    ... )
+
+    """
+
+    body = {}
+    if mappings:
+        if _get_distribution(client) == 'opensearch' or _get_version_major(client) >= 7:
+            body['mappings'] = mappings  # doc type deprecated
+        else:
+            if doc_type:
+                body['mappings'] = {doc_type: mappings}
+            else:
+                body['mappings'] = {index: mappings}
+    if settings:
+        body['settings'] = settings
+    if body == {}:
+        body = None
+    response = client.indices.create(index, body, ignore=[400, 404])
```

Comment:
What is the reason here to ignore `400` & `404`?

Reply:
it's a common pattern to ignore 400 (IndexAlreadyExistsException) when creating an index and 400,404 (IndexNotFoundError) when deleting an index.
Anyway wrangler will log a warning message upon any of these exceptions.

https://elasticsearch-py.readthedocs.io/en/7.x/api.html#ignore

Reply:
I've modified the `create_index` method to only ignore 400